compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -490,13 +490,13 @@ extern "C" LLVMTargetMachineRef LLVMRustCreateTargetMachine(
     assert(ArgsCstrBuff[ArgsCstrBuffLen - 1] == '\0');
     auto Arg0 = std::string(ArgsCstrBuff);
     buffer_offset = Arg0.size() + 1;
-    auto ArgsCppStr =
-        std::string(ArgsCstrBuff + buffer_offset, ArgsCstrBuffLen - 1);
+    auto ArgsCppStr = std::string(ArgsCstrBuff + buffer_offset,
+                                  ArgsCstrBuffLen - buffer_offset);
     auto i = 0;
     while (i != std::string::npos) {
       i = ArgsCppStr.find('\0', i + 1);
       if (i != std::string::npos)
-        ArgsCppStr.replace(i, i + 1, " ");
+        ArgsCppStr.replace(i, 1, " ");
```

Comment:
So, re-reading the reference about `std::string::npos`, I think I may have misunderstood something about it.

Why is it correct at all as part of a conditional expression like a `while` or `if`? It only takes on its magic "until the end of the string" meaning when used as an argument to `std::string`'s functions, doesn't it? Which the control-flow expressions are not invocations of.

Reply:
https://cplusplus.com/reference/string/string/find/

> Return Value
> The position of the first character of the first match.
> If no matches were found, the function returns [string::npos](https://cplusplus.com/string::npos).

`npos` isn't magic, it's in practice just a way to say `-1`. find() returns npos if something isn't found, and at that point we don't take this if / exit the loop.

Reply:
Ah, okay!